### PR TITLE
added integrity analysis event on apiserver

### PR DESCRIPTION
**Pull request by mehab**
### Description
Added integrity analysis event on hyades apiserver side for periodic analysis, manual component creation and bom upload events

### Addressed Issue

Addresses https://github.com/DependencyTrack/hyades/issues/699

### Checklist

- [x] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

File: src/main/java/org/dependencytrack/resources/v1/ComponentResource.java [hunk 1]
```diff
@@ -316,7 +321,15 @@ public Response createComponent(@PathParam("uuid") String uuid, Component jsonCo
             component.setNotes(StringUtils.trimToNull(jsonComponent.getNotes()));
 
             component = qm.createComponent(component, true);
-            kafkaEventDispatcher.dispatchBlocking(new ComponentRepositoryMetaAnalysisEvent(component));
+            ComponentProjection componentProjection =
```
Comment (VithikaS) [inline src/main/java/org/dependencytrack/resources/v1/ComponentResource.java:324]: purl coordinates will not have the qualifier information which will be required for integrity data. We would need the canonicalized form of purl i think.

Comment in Component model:

 `private String purlCoordinates; // Field should contain only type, namespace, name, and version. Everything up to the qualifiers`

File: src/main/proto/org/hyades/repometaanalysis/v1/repo_meta_analysis.proto [hunk 2]
```diff
@@ -11,6 +11,15 @@ option java_package = "org.hyades.proto.repometaanalysis.v1";
 message AnalysisCommand {
   // The component that shall be analyzed.
   Component component = 1;
+  FetchMeta fetch_meta = 2;
+
+}
+
+enum FetchMeta{
+  FETCH_META_UNSPECIFIED = 0;
```
Comment (sahibamittal) [inline src/main/proto/org/hyades/repometaanalysis/v1/repo_meta_analysis.proto:19]: Do we need FETCH_META_UNSPECIFIED? as it should be either one or both but never unspecified. If it is unspecified, the event wouldn't be called at all.

Comment (mehab) [reply src/main/proto/org/hyades/repometaanalysis/v1/repo_meta_analysis.proto:19]: It is part of the proto enum standard. Would not satisfy the enum standard without the unspecified enum field

**Review by sahibamittal — APPROVED** [review]
LGTM👍🏼

**Review by sahibamittal — APPROVED** [review]

**Pull request merged**